core/src/main/java/org/bitcoinj/evolution/SimplifiedMasternodeListManager.java

```diff
@@ -91,6 +91,9 @@ protected boolean removeEldestEntry(Map.Entry<Sha256Hash, SimplifiedQuorumList>
 
     public SimplifiedMasternodeListManager(Context context) {
         super(context);
+        if (Runtime.getRuntime().totalMemory() <= Math.pow(1024, 3)) {
+            MAX_CACHE_SIZE = 1;
+        }
```

Comment:
Runtime.getRuntime().totalMemory() does not appear to get the total amount of device memory. 
It returned 256 MB on a 16 GB machine and 8MB on a 1 GB Krip phone.

besides that, MAX_CACHE = 1 appears to work as far as instantsend/chainlock functionality.  I think we should make MAX_CACHE = 1 as static final since keeping 8 or 10 lists in memory on other higher memory devices is unnecessary.  Therefore there is no need to check for device memory size.  A separate task will be created to perhaps keep 2 lists in memory (the 8th previous and the 7th previous) or some other optimization for higher memory devices.  This would address issues where a chainlock or instantsendlock arrives before the DMN List is updated.

Reply:
I will do this change.